drivers/hmis/app/models/hmis/user_hmis_data_source_role.rb

```diff
@@ -6,7 +6,30 @@
 
 class Hmis::UserHmisDataSourceRole < ::ApplicationRecord
   self.table_name = :user_hmis_data_source_roles
+  has_paper_trail(
+    meta: {
+      referenced_user_id: :referenced_user_id,
+      referenced_entity_name: :referenced_entity_name,
+    },
+  )
+
   belongs_to :user
   belongs_to :role
   belongs_to :data_source, class_name: 'GrdaWarehouse::DataSource'
+
+  def referenced_user_id
+    user.id
+  end
+
+  def referenced_entity_name
+    role.name
+  end
+
+  def self.describe_changes(version, _changes)
+    if version.event == 'create'
+      ["Added role #{version.referenced_entity_name}"]
```

Comment:
Isn't `version.referenced_entity_name` going to return the name of the role, not the user?  So this'll say "Added role to Admin Role" instead of "Added role Admin Role to Elliot"?

Reply:
It would be "Added role Admin Role" and "Removed role Admin Role," which could be fine if it's displayed in the context of a user? This is fine by me, we can revisit the wording if needed when we get to displaying it.

Reply:
Got it, so this is a change to a user, indicating role addition or removal.  I think that's fine as is.